src/foam/nanos/logger/DiscardLogger.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.logger',
+  name: 'DiscardLogger',
+  implements: [ 'foam.nanos.logger.Logger' ],
+  extends: [ 'ProxyLogger' ],
+
+  properties: [
+    {
+      name: 'discardInfo',
+      class: 'Boolean',
+      value: 'false'
+    },
+    {
+      name: 'discardWarning',
+      class: 'Boolean',
+      value: 'false'
+    },
+    {
+      name: 'discardError',
+      class: 'Boolean',
+      value: 'false'
+    },
+    {
+      name: 'discardDebug',
+      class: 'Boolean',
+      value: 'false'
+    }
+  ],
+    
+  methods: [
+    {
+    name: 'info',
+    args: [
+      {
+      name: 'message',
+      javaType: 'String'  
+      }
+    ],
+    javaCode: `
```

Comment:
the javaCode is just the method implementation so between
public void info(Object... args) {
// and 
}

so just the 
  if ( ! getDiscardInfo() ) {
    getDelegate().info(args);
  }